src/Umbraco.Web/PropertyEditors/ValueConverters/MemberPickerValueConverter.cs

```diff
@@ -27,44 +30,77 @@ public override PropertyCacheLevel GetPropertyCacheLevel(IPublishedPropertyType
             => PropertyCacheLevel.Snapshot;
 
         public override Type GetPropertyValueType(IPublishedPropertyType propertyType)
-            => typeof (IPublishedContent);
+        {
+            var isMultiple = IsMultipleDataType(propertyType.DataType);
+            return isMultiple
+                    ? typeof(IEnumerable<IPublishedContent>)
+                    : typeof(IPublishedContent);
+        }
+
+        private bool IsMultipleDataType(PublishedDataType dataType)
+        {
+            var config = ConfigurationEditor.ConfigurationAs<MemberPickerConfiguration>(dataType.Configuration);
+            return config.Multiple;
+        }
 
         public override object ConvertSourceToIntermediate(IPublishedElement owner, IPublishedPropertyType propertyType, object source, bool preview)
         {
-            var attemptConvertInt = source.TryConvertTo<int>();
-            if (attemptConvertInt.Success)
-                return attemptConvertInt.Result;
-            var attemptConvertUdi = source.TryConvertTo<Udi>();
-            if (attemptConvertUdi.Success)
-                return attemptConvertUdi.Result;
-            return null;
+            var nodeIds = source.ToString()
+                .Split(new[] { "," }, StringSplitOptions.RemoveEmptyEntries)
+                .Select(Udi.Parse)
```

Comment:
@bjarnef Hmm, previously we could be getting an `int` or a `Udi`, now you're just assuming this is always a `Udi`, that can't be right?

Reply:
I know it will be a `Udi` list for newly picked values but if people already have a single `int` identifier stored then we break that here.

Reply:
Yeah, I think I just was inspired from the media picker https://github.com/umbraco/Umbraco-CMS/blob/51c1234db3d747cdb609236daabb5625f8058da0/src/Umbraco.Web/PropertyEditors/ValueConverters/MediaPickerValueConverter.cs#L59-L62 ... but I guess this already existed since Media Picker V2.

Reply:
@nul800sebastiaan not sure if this is a left over from previous when member picker V1 stored id, but when the new pickers was refactored in the V2 pickers Content Picker, Media Picker, Media Picker, MNTP, Member Picker stored an `Udi` instead. In Umbraco v8 the "2" in the naming was removed.

Hasn't member picker stored an `Udi` since the initial release of Umbraco v8?

Reply:
It looks like the `{"idType", "udi"}` has been there for a while.

https://github.com/umbraco/Umbraco-CMS/commit/a97900515b0e7bbfba6e7d32d8175505ab2bf863

Reply:
I have tested with Umbraco v8.0.0 where it also stored an `Udi`, so I think it is an oversight this part wasn't changed like in Content Picker and Media Picker value converters.

![image](https://user-images.githubusercontent.com/2919859/88385546-d59c1100-cdae-11ea-8574-a4dbe8d58d7c.png)

Reply:
Sure, but there's also people migrating from v7 which is why I'm assuming that int still needs to be there..

Reply:
@nul800sebastiaan sure, but isn't it an issue with Media Picker as well? https://github.com/umbraco/Umbraco-CMS/blob/v8/contrib/src/Umbraco.Web/PropertyEditors/ValueConverters/MediaPickerValueConverter.cs#L59-L63

or is it what this migration handle? https://github.com/umbraco/Umbraco-CMS/blob/2f978e96d4fbf84d8786c3245566c8800916a74b/src/Umbraco.Core/Migrations/Upgrade/V_8_0_0/UpdatePickerIntegerValuesToUdi.cs